Makefile

```
@@ -0,0 +1,61 @@
+.PHONY: all check-env docker docker-dev docker-tools dev lint dep build test mock protobuf changelog clean 
+
+all: clean test build docker
+
+MAJOR_VERSION := $(shell echo $(version) | cut -d . -f 1)
+MINOR_VERSION := $(shell echo $(version) | cut -d . -f 1-2)
+PATCH_VERSION := $(version)
+
+check-env:
+ifndef version
+	$(error version is undefined)
+endif
+
+docker: check-env
+	docker build \
+		--build-arg version=$(PATCH_VERSION) \
+		-t mesg/engine:$(MAJOR_VERSION) \
+		-t mesg/engine:$(MINOR_VERSION) \
+		-t mesg/engine:$(PATCH_VERSION) \
+		-t mesg/engine:latest \
+		.
+
+docker-dev:
+	docker build -t mesg/engine:dev --build-arg version=$(version) .
+
+docker-tools:
+	docker build -t mesg/tools:local -f Dockerfile.tools .
+
+dev: docker-dev
+	- ./scripts/dev.sh
+
+dep:
+	go mod download
+
+build: dep
+	go build -mod=readonly -o ./bin/engine -ldflags="-X 'github.com/mesg-foundation/engine/version.Version=$(version)'" core/main.go
```

Again, Why using $(version) without the check-env?
Here i would suggest to force version:
```suggestion
	check-env
	go build -mod=readonly -o ./bin/engine -ldflags="-X 'github.com/mesg-foundation/engine/version.Version=$(version)'" core/main.go
```